python/pfs/pipe2d/gitChangelog.py

```diff
@@ -36,529 +25,187 @@
 """
 
-GITHUB_API_BASE_URL = r'https://api.github.com/repos/Subaru-PFS'
-"""Base URL for GitHub REST API for querying GitHub API (`str`)
-"""
-
-GITHUB_API_MAX_PAGES = 20
-""" Maximum number of pages retrieved per github api request (`int`).
-When querying the GitHub REST API, the result is
-provided across multiple pages.
-While this must be finite, provide a limit to this.
-"""
-
-GITHUB_API_PARAMS = {'per_page': 100}
-"""Github API request parameters (`dict`)
-Limit the number of results per page in a REST API query to the above.
+GITHUB_ORGANIZATION = 'Subaru-PFS'
+"""Name of organization as specified in Github (`str`)
 """
 
-RATE_LIMIT_URL = r'https://api.github.com/rate_limit'
-"""The github request rate limit URL (`str`).
-The allowed number of GITHUB API requests per hour for the user in question.
-This is used to check if the number of requests have been exceeded so
-that situation can be handled
-(that is, an error message is raised and the program exits)
+MERGE_MESSAGE_REGEX = re.compile(r"^Merge branch \'tickets\/([a-zA-Z0-9]+\-[0-9]+)\'$")  # noqa: E501
+"""Merged message regular expression (`re.Pattern`).
+The regular expression of the standard message generated when a ticket branch
+is merged to master.
 """
 
-GITHUB_API_TICKET_REGEX = r'^Subaru-PFS:tickets/([A-Z0-9]+-[0-9]+)$'
-"""JIRA ticket regular expression (`str`).
-The JIRA ticket name is stored in the GitHub API in the above
-format. This is used to parse the ticket number.
-"""
-
-TAG_REGEX = r"^([0-9]+)\.([0-9]+)(\.[0-9]+)?$"
-"""Git tag regular expression (`str`).
-The format of a tag. In the above case, M.m[.p] for releases.
+TAG_REGEX = re.compile(r"^w\.(\d{4})\.(\d{2})$")
+"""Git tag regular expression (`re.Pattern`).
+The format of a tag. In the above case, for weeklies eg w.2022.17 .
 """
 
 NOT_TAGGED = 'NOT-TAGGED'
 """Label for tickets not assigned to a tag (`str`).
 For grouping recently closed tickets that are not in a release, so not tagged.
 """
 
-DATE_TOL_MIN = 2
-"""The tolerance for comparing ticket timestamps against tag ranges [minutes]
-"""
-
-REPO_PRINCIPAL = 'pfs_pipe2d'
-"""Name of repo that provides tag-date information
-"""
-
-
-def _no_authentication(authfile):
-    """Handles situation where no GitHub API authentication
-    is to be provided.
 
-    Parameters
-    ----------
-    authfile : `str`
-        File containing authentication credentials
+def readAccessToken():
+    """Get access token for GitHub from file.
 
     Returns
     -------
-    auth : `tuple` (`set`, `set`)
-        tuple to enable HTTP authentication
+    authToken : `str`
+        Access token
     """
-    logger.debug('Using no authentication')
-    return None
+    logger.debug(f'Reading access token from {GITHUB_AUTH}..')
+    with open(GITHUB_AUTH) as fd:
+        return fd.readline().strip()
 
 
-def _external_authentication(authfile):
-    """Handles authentication being provided by an external file.
+def getCommitMessage(commit):
```

Comment:
Since you're doing a rewrite, can I encourage you to include typing information?